internal/configconverter/otlp_insecure_mover.go

```diff
@@ -0,0 +1,41 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configconverter
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+
+	"go.opentelemetry.io/collector/config/configparser"
+)
+
+func MoveOTLPInsecureKey(in *configparser.ConfigMap) *configparser.ConfigMap {
+	const expr = "exporters::otlp(/\\w+)?::insecure"
+	insecureRE, _ := regexp.Compile(expr)
+	out := configparser.NewConfigMap()
+	for _, k := range in.AllKeys() {
+		v := in.Get(k)
+		match := insecureRE.FindStringSubmatch(k)
+		if match == nil {
+			out.Set(k, v)
+		} else {
+			tlsKey := fmt.Sprintf("exporters::otlp%s::tls::insecure", match[1])
+			log.Printf("Unsupported key found: %s. Moving to %s\n", k, tlsKey)
```

Comment:
Maybe we should change the wording to encourage user to update the config so we can drop it later on?

Something like:
```
[DEPRECATION NOTICE] Configuration parameter "exporters -> otlp -> insecure" is deprecated. 
Please move it to "exporters -> otlp -> tls -> insecure".
More details: https://github.com/open-telemetry/opentelemetry-collector/pull/4063/
```

@emaderer thoughts?